### PR TITLE
Remove 2 BAT limit for uphold

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/24759

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/resources/android_page/components/connect_wallet_modal.tsx [hunk 1]
```diff
@@ -99,51 +89,16 @@ export function ConnectWalletModal (props: Props) {
...
-        props.onContinue(selectedProvider.type)
-      }
-    }
-
-    if (!selectedProvider) {
-      return null
-    }
-
-    return (
-      <style.minimumBalanceWarning>
```
Comment (zenparsing) [inline components/brave_rewards/resources/android_page/components/connect_wallet_modal.tsx:115]: Can we now remove the `minimumBalanceWarning` style exported from `connect_wallet_modal.style.ts`?

File: components/brave_rewards/resources/page/components/connect_wallet_modal.tsx [hunk 2]
```diff
@@ -126,26 +117,6 @@ export function ConnectWalletModal (props: Props) {
...
-                <style.minimumBalanceWarning>
-                  {
-                    formatMessage(getString('minimumBalanceWarning'), [
-                      selectedProvider.name,
-                      getMinimumBalance(selectedProvider.type)
-                    ])
-                  }
-                  <div className='login'>
-                    <a
-                      data-test-id='connect-login-button'
```
Comment (zenparsing) [inline components/brave_rewards/resources/page/components/connect_wallet_modal.tsx:140]: I think this element is referenced from a Rewards browser test. We can probably remove that test now. (`UpholdLimitNoBAT` maybe?)

File: browser/ui/webui/brave_webui_source.cc [hunk 3]
```diff
@@ -768,8 +768,6 @@ void CustomizeWebUIHTMLSource(const std::string &name,
         { "connectWalletInfoListItem1", IDS_BRAVE_REWARDS_CONNECT_WALLET_INFO_LIST_ITEM_1 },  // NOLINT
         { "connectWalletInfoListItem2", IDS_BRAVE_REWARDS_CONNECT_WALLET_INFO_LIST_ITEM_2 },  // NOLINT
         { "continue", IDS_BRAVE_REWARDS_CONTINUE },
-        { "continueToLogin", IDS_BRAVE_REWARDS_CONTINUE_TO_LOGIN },
-        { "minimumBalanceWarning", IDS_BRAVE_REWARDS_MINIMUM_BALANCE_WARNING },
```
Comment (emerick) [inline browser/ui/webui/brave_webui_source.cc:772]: Do we need to remove the associated strings in the .grd file as well?

**Review by emerick — APPROVED** [review]
LGTM, one minor comment

**Review by zenparsing — APPROVED** [review]
Looks great!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.45.38 Chromium: 105.0.5195.102 (Official Build) nightly (64-bit)
-- | --
Revision | 4c16f5ffcc2da70ee2600d5db77bed423ac03a5a-refs/branch-heads/5195_55@{#4}
OS | Windows 11 Version 21H2 (Build 22000.918)
```

Went through the Uphold login process on a new profile and ensured there weren't any errors indicating that `2 BAT` was needed before being able to signup for an account as per the following:

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
---------------|----------------|-----------------|----------------|----------------|-----------------

**Pull request merged**